### DISCUSSION TITLE
ng-tech-events has a UI now and a dummy REST/JSON API🚀 🎉 🎉

### DISCUSSION THREAD

**Discussion by kaf-lamed-beyt** [kaf-lamed-beyt/ng-tech-events#58] [Announcements]
First off! I need to talk about how I almost cried because Git decided to trick me! I tried setting up husky for pre-commits hooks. Long story short I got confused and I went on to initialize a git repo inside another directory. 

I wan craze! 😩 

Feel free to go through the `/frontend` dir and see if there is anything that'd need improving, now or  in the future

Anyhow sha... I've modified all that needs to be modified, and we're live &mdash; locally for now, at least.

Going forward, we'll proceed by improving the UI. For now, here are some of the things that needs to be covered.

- First off, is the image for each event. We need to make them unique. I already have an idea of how to achieve that with [Satori](https://github.com/vercel/satori), Vercel's Engine which lets you convert HTML and CSS to SVG images.

- dynamic routes for each event such that when it is clicked upon, people are taken to the respective page with a corresponding UI that entails the information about the event.

- search functionality for all the events, filtered and unfiltered.

- The **"add new event"** button can just render a modal that shows a text accompanied by a button that points people to this repo. Although, in the design, I discussed with the designer that we need to have a form that takes input from users. The aim is to send the data from the form as JSON to a file in the repo, somewhere. 

    I'm still experimenting with this. Hopefully, I don't end up... yunno? 😉 

- There's a pattern that she followed with the event cards. We established a concept that'd help people distinguish event types based on their categories hence, the reason why the gradient border on each event card appears different. An ideal approach would be to allow the event card to accept a prop whose value is passed to the `backgroundImage` property since border properties can not be assigned gradient values.

Here's the [link to the design](https://www.figma.com/file/GYPAOIVSjQmos9RSWg0NKa/Techie-assemble?type=design&node-id=346-1026&mode=design&t=xXNszugS8RBn2eN1-0). Yes. It is meant to be this plain. Kindly go through it.

I'll start creating issues with respect to the items I've listed above. 

One last thing, kindly go through the README file in the `/frontend` dir too. It kinda walks you through the project structure, and some ideas behind why some things are the way they are.

cc: @Jemeni11 @orionEsu @PrimeFord @Petsamuel @yhoungdev @orionEsu @DeveloperAspire @Vida-TG @Tonie-NG @Opeyimika-sudo

**Comment by PrimeFord** [comment]
okay chief got it..... Thanks for having me by the way

**Reply by kaf-lamed-beyt** [reply]
Anytime, chief!!

**Comment by Petsamuel** [comment]
Nice nice, weldone 👏

I hv ran it locally, no be small work you do.
Tag me to the designer twitter handle

**Reply by kaf-lamed-beyt** [reply]
Ayy! Thanks chief! 🙌🏽

Issues should be out this night or tomorrow I've been pretty occupied. 

The link to her handle is in the footer, actually. But [here it is](https://twitter.com/_mercyidowu?t=sDKKBKcTXMV2lLMVgXNALg&s=09)

**Comment by orionEsu** [comment]
This is lovely Man. Let's get working

**Comment by Jemeni11** [comment]
Wow this is impressive :fire:

**Reply by kaf-lamed-beyt** [reply]
Open issues o 🤲🏽
